packages/meteor-gazelle-createdby/lib/behavior/behavior.js

```diff
@@ -0,0 +1,109 @@
+// Based on the meteor-astronomy-timestamp-behavior package:
+// https://github.com/jagi/meteor-astronomy-timestamp-behavior/
+Astro.createBehavior({
+  name: 'createdby',
+  options: {
+    hasCreatedByField: true,
+    createdByFieldName: 'createdBy',
+    hasUpdatedByField: true,
+    updatedByFieldName: 'updatedBy',
+    hasCreatedAtField: true,
+    createdAtFieldName: 'createdAt',
+    hasUpdatedAtField: true,
+    updatedAtFieldName: 'updatedAt'
+  },
+  validators: {
+    'hasCreatedByField': Validators.boolean(),
+    'createdByFieldName': Validators.string(),
+    'hasUpdatedByField': Validators.boolean(),
+    'updatedByFieldName': Validators.string(),
+    'hasCreatedAtField': Validators.boolean(),
+    'createdAtFieldName': Validators.string(),
+    'hasUpdatedAtField': Validators.boolean(),
+    'updatedAtFieldName': Validators.string()
+  },
+  events: {
+    addbehavior: function (behaviorData) {
+
+      // Set default behavior's options if they were not provided in the schema.
+      if (_.isUndefined(behaviorData.hasCreatedByField)) {
+        behaviorData.hasCreatedByField = behavior.options.hasCreatedByField;
+      }
+      if (_.isUndefined(behaviorData.CreatedByFieldName)) {
+        behaviorData.createdByFieldName = behavior.options.CreatedByFieldName;
+      }
+      if (_.isUndefined(behaviorData.hasUpdatedByField)) {
+        behaviorData.hasUpdatedByField = behavior.options.hasUpdatedByField;
+      }
+      if (_.isUndefined(behaviorData.updatedByFieldName)) {
+        behaviorData.updatedByFieldName = behavior.options.updatedByFieldName;
+      }
+      if (_.isUndefined(behaviorData.hasCreatedAtField)) {
+        behaviorData.hasCreatedAtField = behavior.options.hasCreatedAtField;
+      }
+      if (_.isUndefined(behaviorData.CreatedAtFieldName)) {
+        behaviorData.createdAtFieldName = behavior.options.CreatedAtFieldName;
+      }
+      if (_.isUndefined(behaviorData.hasUpdatedAtField)) {
+        behaviorData.hasUpdatedAtField = behavior.options.hasUpdatedAtField;
+      }
+      if (_.isUndefined(behaviorData.updatedAtFieldName)) {
+        behaviorData.updatedAtFieldName = behavior.options.updatedAtFieldName;
+      }
+
+      // Check validity of options.
+      checks.behaviorData.call(this, behaviorData);
```

Comment:
I'm removing this. It's from the `meteor-astronomy-timestamp-behavior` package and I was unsure about it.